### PR TITLE
🐛 Fixed windows CI not reporting exitcodes correctly

### PR DESCRIPTION
**Pull request by AnotherFoxGuy**
This uses https://github.com/ilammy/msvc-dev-cmd to setup the MSVC environment, and https://github.com/AnotherFoxGuy/conan-onefile for the installation of conan

-----------------------------------

Pull request checklist:
- [x] `CHANGELOG.md` was updated, if applicable

### CODE REVIEW

**Comment by pajlada** [comment]
`ilammy/msvc-dev-cmd@v1.6.0` seems like a nice solution for this, but I'll keep it for after the cmake PR has been merged into master

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -63,59 +63,49 @@ jobs:
...
 
-      - name: Install dependencies (Windows)
+      - name: Download conan
         if: startsWith(matrix.os, 'windows')
         run: |
-            choco install conan -y
-
-            refreshenv
-        shell: cmd
+            Invoke-WebRequest -Uri "https://github.com/AnotherFoxGuy/conan-onefile/releases/latest/download/conan.exe" -OutFile "C:\vcpkg\conan.exe"
```
Comment (pajlada) [inline .github/workflows/build.yml:76]: Let's revert these changes and keep install conan through choco if possible

**Review by pajlada — CHANGES_REQUESTED** [review]

File: .github/workflows/build.yml (hunk 1 above)
Comment (AnotherFoxGuy) [reply .github/workflows/build.yml:76]: The reason I changed it is because installing conan using choco takes anywhere between 3 to 8 minutes, and this method only takes 5 seconds
Also, this removes the need for a second "Add conan to path" action

Comment (pajlada) [reply .github/workflows/build.yml:76]: Yeah let's leave that for another time though - the windows CI is still not the one we have to wait for in the github actions

Comment (pajlada) [reply .github/workflows/build.yml:76]: maybe we could just cache the choco results instead?

Comment (pajlada) [reply .github/workflows/build.yml:76]: either way something for another PR

**Review by pajlada — APPROVED** [review]
